src/main/java/com/iota/iri/MilestoneTracker.java

```diff
@@ -0,0 +1,549 @@
+        (new Thread(() -> {
+            log.info("Initializing Ledger Validator...");
+            try {
+                ledgerValidator.init();
+                ledgerValidatorInitialized.set(true);
+            } catch (Exception e) {
+                log.error("Error initializing snapshots. Skipping.", e);
+            }
+            log.info("Tracker started.");
+            while (!shuttingDown) {
+                long scanTime = System.currentTimeMillis();
+
+                try {
+                    if(latestSolidSubtangleMilestoneIndex < latestMilestoneIndex) {
+                        updateLatestSolidSubtangleMilestone();
+                    }
+
+                    Thread.sleep(Math.max(1, RESCAN_INTERVAL - (System.currentTimeMillis() - scanTime)));
+                } catch (final Exception e) {
+                    log.error("Error during Solid Milestone updating", e);
+                }
+            }
+        }, "Solid Milestone Tracker")).start();
+    }
+
+    private void spawnMilestoneSolidifier() {
+        new Thread(() -> {
+            while(!shuttingDown) {
+                unsolidMilestones.forEach((milestoneHash, milestoneIndex) -> {
+                    try {
+                        // remove old milestones that are not relevant anymore
+                        if(milestoneIndex <= latestSolidSubtangleMilestoneIndex) {
+                            unsolidMilestones.remove(milestoneHash);
+                        }
+
+                        // remove milestones that have become solid and are in our check range
+                        else if(milestoneIndex < latestSolidSubtangleMilestoneIndex + MILESTONE_SOLIDIFY_AHEAD_RANGE && transactionValidator.checkSolidity(milestoneHash, true)) {
+                            unsolidMilestones.remove(milestoneHash);
+                        }
+                    } catch(Exception e) {
+                        log.error("Error while trying to solidify milestones", e);
+                    }
+                });
+
+                try { Thread.sleep(500); } catch (InterruptedException e) { /* do nothing */ }
+            }
+        }, "Milestone Solidifier").start();
+    }
+
+    /**
+     * This method analyzes a transaction to determine if it is a milestone.
+     *
+     * In contrast to {@link #validateMilestone} it also updates the internal variables keeping track of the latest
+     * milestone and dumps a log message whenever a change is detected. So this method can be seen as the core logic for
+     * scanning and keeping track of the latest milestones. Since it internally calls {@link #validateMilestone} a
+     * milestone will have its corresponding {@link MilestoneViewModel} created after being processed by this method.
+     *
+     * The method first checks if the transaction originates from the coordinator address, so the check is relatively
+     * cheap and can easily performed on any incoming transaction.
+     *
```

Comment:
Great javadocs!
The only problem is that they are a duplication of the javadocs just below.

Use the `@see` annotation so you won't have to change in 2 places everytime.
e.g.
```
 /**
     * This method analyzes a transaction to determine if it is a milestone.
     *
     * @see #analyzeMilestoneCandidate(TransactionViewModel)
     * @param potentialMilestoneTransactionHash hash of the transaction that shall get checked for being a milestone
     * @return VALID if the transaction is a milestone, INCOMPLETE if the bundle is not complete and INVALID otherwise
     * @throws Exception if something goes wrong while retrieving information from the database
     */
```

You still repeating some stuff but I don't know of a better solution.